src/style/style.js

```diff
@@ -46,12 +53,47 @@ const ignoredDiffOperations = util.pick(diff.operations, [
     'setPitch'
 ]);
 
+export type StyleOptions = {
+    validate?: boolean,
+    localIdeographFontFamily?: string
+};
+
+type ZoomHistory = {
+    lastIntegerZoom: number,
+    lastIntegerZoomTime: number,
+    lastZoom: number
+};
+
 /**
  * @private
  */
 class Style extends Evented {
-
-    constructor(stylesheet, map, options) {
+    map: Map;
+    stylesheet: StyleSpecification;
+    animationLoop: AnimationLoop;
+    dispatcher: Dispatcher;
+    sprite: ImageSprite;
+    spriteAtlas: SpriteAtlas;
+    lineAtlas: LineAtlas;
+    glyphSource: GlyphSource;
+    light: Light;
+
+    _layers: {[string]: StyleLayer};
+    _order: Array<string>;
+    sourceCaches: {[string]: SourceCache};
+    zoomHistory: ZoomHistory | {};
```

Comment:
Is this typed as `ZoomHistory | {}` rather than `?ZoomHistory` to reduce invasiveness of the change?  ( 👍 if so, just making sure )